BlueprintUI/Sources/Element/ElementContent.swift

```diff
@@ -72,17 +95,34 @@ extension ElementContent {
     public init(intrinsicSize: CGSize) {
         self = ElementContent(measureFunction: { _ in intrinsicSize })
     }
-
 }
 
+extension ElementContent {
+    public init(child: Element, environment environmentAdapter: @escaping (inout Environment) -> Void) {
+        self.storage = EnvironmentAdaptingStorage(
+            layout: PassthroughLayout(),
+            adapter: environmentAdapter,
+            child: child)
+    }
 
-fileprivate protocol ContentStorage: Measurable {
-    var childCount: Int { get }
-    func performLayout(attributes: LayoutAttributes) -> [(identifier: ElementIdentifier, node: LayoutResultNode)]
+    public init<K>(child: Element, key: K.Type, value: K.Value) where K: EnvironmentKey {
```

Comment:
Nit: Don't love the single letter generics. `Key`?

Reply:
Sure